app/main.py

```diff
@@ -1,25 +1,71 @@
 from abc import ABC
+from typing import Callable
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: object, name: str) -> None:
+        self.public_name = name
+        self.protected_name = "_" + name
+
+    def __get__(self, instance: object, owner: object) -> int:
+        return getattr(instance, self.protected_name)
+
+    def __set__(self, instance: object, value: int) -> None:
+        if not self.min_amount <= value <= self.max_amount:
+            raise ValueError(f"Value should be greater than {self.min_amount} "
+                             f"and greater than {self.max_amount}")
+        setattr(instance, self.protected_name, value)
 
 
 class Visitor:
-   pass
+    def __init__(self, name: str, age: int, weight: int, height: int) -> None:
+        self.name = name
+        self.age = age
+        self.weight = weight
+        self.height = height
 
 
 class SlideLimitationValidator(ABC):
-    pass
+    def __init__(self, age: int, weight: int, height: int) -> None:
+        self.age = age
+        self.weight = weight
+        self.height = height
 
 
 class ChildrenSlideLimitationValidator(SlideLimitationValidator):
-    pass
+    age = IntegerRange(4, 14)
+    height = IntegerRange(80, 120)
+    weight = IntegerRange(20, 50)
+
+    def __init__(self, age: int, height: int, weight: int) -> None:
+        super().__init__(age=age, height=height, weight=weight)
 
 
 class AdultSlideLimitationValidator(SlideLimitationValidator):
-    pass
+    age = IntegerRange(14, 60)
+    height = IntegerRange(120, 220)
+    weight = IntegerRange(50, 120)
+
+    def __init__(self, age: int, height: int, weight: int) -> None:
+        super().__init__(age=age, height=height, weight=weight)
 
 
 class Slide:
-    pass
+    def __init__(self, name: str, limitation_class: Callable) -> None:
```

Comment:
`limitation_class` is always the subclass of SlideLimitationValidator class. You must change typehint.

Reply:
<img width="429" alt="image" src="https://user-images.githubusercontent.com/77115535/228946002-68243cdc-3068-4134-a230-ebbf977e7f78.png">
what I have to do with this?

Reply:
You must use `type[SlideLimitationValidator]` instead of `SlideLimitationValidator` as type hint in this case.

Reply:
thanks)